cli/crates/cli/src/cli_input/graph_ref_no_branch.rs

```diff
@@ -0,0 +1,71 @@
+use std::{fmt, str};
+
+/// Parsed project reference. A project reference is a string of the form `account/project@branch`.
+#[derive(Clone, Debug)]
+pub struct GraphRefNoBranch {
+    account: String,
+    graph: String,
+}
+
+impl GraphRefNoBranch {
+    pub(crate) const ARG_DESCRIPTION: &'static str = r#"Graph reference following the format "account/graph""#;
+
+    pub(crate) fn into_parts(self) -> (String, String) {
+        (self.account, self.graph)
+    }
+}
+
+impl str::FromStr for GraphRefNoBranch {
+    type Err = &'static str;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        const GENERIC_ERR: &str =
+            r#"Invalid graph reference. The graph reference argument must follow the format: "account/graph""#;
+
+        let Some((account, graph)) = s.split_once('/') else {
+            return Err(GENERIC_ERR);
+        };
+
+        if account.is_empty() {
+            return Err("The account name is missing before '/'.");
+        }
+
+        if graph.is_empty() {
+            return Err("The graph name is missing.");
+        }
+
+        Ok(GraphRefNoBranch {
```

Comment:
We could error if we encounter a `@` here, otherwise we'll consider that part of the project name.